UnitySDK/Assets/ML-Agents/Scripts/DecisionRequester.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using Barracuda;
+using MLAgents.Sensor;
+using UnityEngine.Serialization;
+
+namespace MLAgents
+{
+    public class DecisionRequester : MonoBehaviour
+    {
+        public int DecisionPeriod = 5;
+        public bool RepeatAction = true;
+        private Agent m_Agent;
+        public void Awake()
+        {
+            m_Agent = gameObject.GetComponent<Agent>();
+            Academy.Instance.AgentSetStatus += MakeRequests;
+        }
+        void OnDestroy()
+        {
+            Academy.Instance.AgentSetStatus -= MakeRequests;
+        }
+        void MakeRequests(int count)
+        {
+            if (count % DecisionPeriod == 0)
```

Comment:
> I know it's "new" functionality, but I really think the ability to stagger which agents request each step would be really useful for developers with lots of agents. Otherwise you get large spikes every DecisionPeriod steps.

This is a good candidate for thinking of jobification of ML-Agents.  Not for 1.0, but later.